### PR TITLE
pybind/ceph_argparse: print flag name before CephChoices in help descs

### PR DESCRIPTION
**Pull request by adk3798**
Fixes: https://tracker.ceph.com/issues/49757

Signed-off-by: Adam King <adking@redhat.com>

Really just wanted to fix the --format flag issue mentioned in the tracker issue but the help commands are auto generated so it's hard to fix them individually. Not sure if this change is a good idea since it affects the help description of so many commands but I'm putting the idea out there.

old help output (all commands): https://pastebin.com/raw/zq3rDj58
help output with this change (all commands): https://pastebin.com/raw/GUwK9nL4

change to orch ls (as an example):

old:`orch ls [<service_type>] [<service_name>] [--export] [plain|json|json-pretty|yaml] [--refresh] `

new:`orch ls [<service_type>] [<service_name>] [--export] [--format {plain|json|json-pretty|yaml}] [--refresh]`

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/ceph_argparse.py [hunk 1]
```diff
@@ -830,8 +830,10 @@ def helpstr(self):
         """
         if self.t == CephBool:
             chunk = "--{0}".format(self.name.replace("_", "-"))
-        elif self.t == CephPrefix or self.t == CephChoices:
+        elif self.t == CephPrefix:
             chunk = str(self.instance)
+        elif self.t == CephChoices:
+            chunk = "--{0} ".format(self.name) + '{' + str(self.instance) + '}'
```
Comment (tchaikov) [inline src/pybind/ceph_argparse.py:836]: nit,
```suggestion
            chunk = f"--{self.name} {{{self.instance}}}'
```

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-03-14_09:40:23-rados-wip-kefu-testing-2021-03-14-1612-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-03-15_02:29:57-rados-wip-kefu-testing-2021-03-15-0034-distro-basic-smithi/

failures are not related.

**Comment by sebastian-philipp** [comment]
When looking at the diff between those two; I noticed:

Before:
```
device light on|off <devid> [ident|fault] [--force]
```

After:
```
device light --enable {on|off} <devid> [--light_type {ident|fault}] [--force] 
```

Which looks somewhat broken as there are positional arguments *after* argument flags.

**Comment by sebastian-philipp** [comment]
Also:

```
Before:
mon feature ls [--with-value]
After:
mon feature ls [--with_value {--with-value}]
```

**Comment by sebastian-philipp** [comment]
```
orch daemon start|stop|restart|reconfig <name>
orch daemon --action {start|stop|restart|reconfig} <name>
```

**Comment by tchaikov** [comment]
@sebastian-philipp you are right. didn't realize that `format` was an exception in this area. will create a PR reverting this change.

**Comment by tchaikov** [comment]
#40111

**Pull request merged**